src/main/java/organice/logic/commands/ProcessingMarkDoneCommand.java

```diff
@@ -0,0 +1,104 @@
+package organice.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static organice.logic.commands.MatchCommand.match;
+
+import organice.model.Model;
+import organice.model.person.Donor;
+import organice.model.person.Nric;
+import organice.model.person.Patient;
+import organice.model.person.TaskList;
+import organice.model.person.exceptions.PersonNotFoundException;
+
+/**
+ * Mark a task on the checklist for the donor and patient to prepare for the organ transplant.
+ * Keyword matching is case insensitive.
+ */
+public class ProcessingMarkDoneCommand extends Command {
+
+    public static final String COMMAND_WORD = "processingMarkDone";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Given the pair of donor and patient identified "
+            + "by their NRICs respectively, this command would mark the task identified as done in their checklist. \n"
+            + "Parameters: ic/PATIENT NRIC ic/DONOR NRIC TASK_NUMBER \n"
+            + "Example: " + COMMAND_WORD + " ic/s4512345A ic/s7711123C 1";
+
+    public static final String MESSAGE_NOT_PROCESSED = "Donor and Patient NRIC must be "
+            + "valid and task number must exist";
+
+    private String firstNricString;
+    private String secondNricString;
+    private String taskNumberString;
+
+    private Nric firstNric;
+    private Nric secondNric;
+    private int taskNumber;
+
+    private Donor donor;
+    private Patient patient;
+
+    private Nric patientNric;
+    private Nric donorNric;
+
+    private TaskList processingList;
+
+    public ProcessingMarkDoneCommand(String firstNricString, String secondNricString, String taskNumberString) {
+        requireNonNull(firstNricString, secondNricString);
+        firstNric = new Nric(firstNricString);
+        secondNric = new Nric(secondNricString);
+        taskNumber = Integer.parseInt(taskNumberString);
```

Comment:
need try catch for `NumberFormatException`

Reply:
amended